### PR TITLE
fix(planner): make aggregator work and add simple stateless tests

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary
Fix degradation due to https://github.com/datafuselabs/databend/pull/5162/ 
- Make aggregation function work properly
- Add simple stateless

## Changelog

- Bug Fix
- Build/Testing/CI

## Related Issues

Fixes #issue

### CODE REVIEW

File: query/src/sql/exec/expression_builder.rs [hunk 1]
```diff
@@ -39,7 +39,12 @@ impl<'a> ExpressionBuilder<'a> {
     }
 
     pub fn build_and_rename(&self, scalar: &Scalar, index: IndexType) -> Result<Expression> {
-        let expr = self.build(scalar)?;
+        let mut expr = self.build(scalar)?;
+        if let Expression::AggregateFunction { .. } = expr {
```
Comment (BohuTANG) [inline query/src/sql/exec/expression_builder.rs:43]: If we want to deal with some other case in the future, `match` is better?

Comment (xudong963) [reply query/src/sql/exec/expression_builder.rs:43]: I'm not sure if we'll add more special cases like `AggregateFunction`. Even if it does, https://github.com/datafuselabs/databend/blob/main/common/planners/src/plan_expression_chain.rs#L73 will prompt us to take care of it

Comment (leiysky) [reply query/src/sql/exec/expression_builder.rs:43]: Will this work with `count(*) + 1`?

Comment (xudong963) [reply query/src/sql/exec/expression_builder.rs:43]: Hold on a second, it's almost ready 🤣

**Review by leiysky — APPROVED** [review]
LGTM

**Pull request merged**